maas/resource_maas_configuration.go

```diff
@@ -0,0 +1,103 @@
+package maas
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceMAASConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS configuration settings through key value pairs. See MAAS server in the [MAAS API documentation](https://maas.io/docs/api) for a full list of keys and their values. Upon destroy, the specified key will no longer be managed by Terraform, but the set value will remain set in MAAS.",
+		CreateContext: resourceMAASConfigurationCreate,
+		ReadContext:   resourceMAASConfigurationRead,
+		UpdateContext: resourceMAASConfigurationUpdate,
+		DeleteContext: resourceMAASConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
```

Comment:
I doubt that we need an import functionality. This is a static value that we set, and we can see with a Terraform state refresh which is the MAAS value for it vs the user's Terraform plan. Most probably we can skip the import.